samples/drawing/drawing.cpp

```diff
@@ -2169,14 +2175,38 @@ void MyFrame::OnSave(wxCommandEvent& WXUNUSED(event))
 #if wxUSE_LIBPNG
                      wxT("PNG image (*.png)|*.png;*.PNG|")
 #endif
+#if wxUSE_SVG
+                     wxT("SVG image (*.svg)|*.svg;*.SVG|")
+#endif
                      wxT("Bitmap image (*.bmp)|*.bmp;*.BMP"),
                      wxFD_SAVE | wxFD_OVERWRITE_PROMPT);
     if (dlg.ShowModal() == wxID_OK)
     {
-        wxBitmap bmp(500, 800);
-        wxMemoryDC mdc(bmp);
-        m_canvas->Draw(mdc);
-        bmp.ConvertToImage().SaveFile(dlg.GetPath());
+        int width, height;
+        m_canvas->GetVirtualSize(&width, &height);
+#if wxUSE_SVG
+        wxString ext = dlg.GetPath().AfterLast('.').Lower();
+        if (ext.Lower() == wxT("svg"))
+        {
+#if wxUSE_GRAPHICS_CONTEXT
+            wxGraphicsRenderer* tempRenderer = m_canvas->GetRenderer();
+            m_canvas->UseGraphicRenderer(NULL);
+#endif
+            wxSVGFileDC svgdc(dlg.GetPath(), width, height, 72, wxT("Drawing sample"));
+            svgdc.SetBitmapHandler(new wxSVGBitmapEmbedHandler());
+            m_canvas->Draw(svgdc);
```

Comment:
This trick with swapping DC's doesn't work if "graphics screen" is displayed. In this case an empty image is saved and artefacts are displayed on the screen. It would be good to disable somehow the option to "save as SVG image" if graphics screen is active.